### PR TITLE
Change mismatched ref input error message

### LINKED ISSUE #202: Better error message for tf.assign
When I try to use tf.assign I'm always running into the error message

TypeError: Input 'ref' of 'Assign' Op requires l-value input

I have no idea what this means. It's a pretty unhelpful error message.

Some context ...

```
a = tf.random_normal([4,4]) # I tried this with tf.constant and tf.Variable and got the same result

tf.assign(tf.slice(a, [1, 0], [-1, -1]), tf.slice(a, [0, 0], [3, -1]))

tf.assign(a, 5)

tf.assign(a, a)
```

All three give the same error message.

### PR DESCRIPTION
**Pull request by irapha**
Closes #202

### CODE REVIEW

File: tensorflow/python/framework/op_def_library.py [hunk 1]
```diff
@@ -618,8 +618,8 @@ def apply_op(self, op_type_name, name=None, **keywords):
         if input_arg.is_ref:
           if not all(x._is_ref_dtype for x in types):  # pylint: disable=protected-access
             raise TypeError(
-                "Input '%s' of '%s' Op requires l-value input" %
-                (input_name, op_type_name))
+                "'%s' Op requires that input '%s' be tf.Variable" %
+                (op_type_name, input_name))
```
Comment (mrry) [inline tensorflow/python/framework/op_def_library.py]: I think this error message could be misleading. Sure, `tf.Variable` is the most common "l-value input", but you can also pass the result of `tf.assign()` to another `tf.assign()` as the l-value, in order to chain updates.

How about instead making the error message:

"'%s' op requires that input '%s' be a mutable tensor (e.g a tf.Variable)"

**Review by mrry — CHANGES_REQUESTED** [review]

File: tensorflow/python/framework/op_def_library.py (hunk 1 above)
Comment (iRapha) [inline tensorflow/python/framework/op_def_library.py]: Done!

**Review by mrry — APPROVED** [review]
LGTM!

**Comment by mrry** [comment]
@tensorflow-jenkins  test this please.

**Pull request merged**